parlai/core/tod/tod_agents.py

```diff
@@ -0,0 +1,787 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+"""
```

Comment:
wow, a lot of detail here! Honestly, this might be more readable in a separate README.md so that it can get formatted in a cleaner manner

Reply:
That's a good call - yeah will do.